src/NUnitFramework/framework/Compatibility/ReflectionExtensions.cs

```diff
@@ -59,7 +59,379 @@ public static Type GetTypeInfo(this Type type)
         {
             return type;
         }
-#elif NETCORE
+    }
+
+#elif PORTABLE
+
+    /// <summary>
+    /// Provides NUnit specific extensions to aid in Reflection
+    /// across multiple frameworks
+    /// </summary>
+    public static class TypeExtensions
```

Comment:
This is the meat of this PR. Tests for this are below. The majority of the other changes are just merging the PORTABLE and NETCORE #defines.